unable to infer FindBugs version using gradle plugin 1.1

I get an exception when trying the latest version of the gradle plugin:

```
Caused by: org.gradle.api.GradleException: Unable to infer the version of FindBugs from currently specified FindBugs classpath: [spotbugs-3.1.0-RC3.jar, asm-debug-all-6.0_ALPHA.jar, jcip-annotations-1.0.jar, dom4j-1.6.1.jar, commons-lang-2.6.jar, jaxen-1.1.6.jar, jdepend-2.9.1.jar, jFormatString-3.0.0.jar, spotbugs-annotations-3.1.0-RC3.jar, xml-apis-1.0.b2.jar, jsr305-3.0.1.jar]
        at com.github.spotbugs.internal.spotbugs.FindBugsClasspathValidator.getFindbugsVersion(FindBugsClasspathValidator.java:43)
        at com.github.spotbugs.internal.spotbugs.FindBugsClasspathValidator.validateClasspath(FindBugsClasspathValidator.java:19)
        at com.github.spotbugs.SpotBugsTask.run(SpotBugsTask.java:208)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:141)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:134)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:123)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:692)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:675)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.execute(ExecuteActionsTaskExecuter.java:115)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.execute(ExecuteActionsTaskExecuter.java:109)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:90)
```

The [pattern](https://github.com/spotbugs/spotbugs/blob/master/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java#L38) doesn't seem to match the spotbugs jars. It should be relatively easy to update the pattern, but the version matching will also need to change as it currently just expects numbers or dots. At the least it would also need to accept the `-RCN` suffix, but I'm not sure if there are others.

--- a/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
+++ b/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
@@ -20,8 +20,8 @@ public class FindBugsClasspathValidator {
       boolean java8orMore = javaVersion.compareTo(JavaVersion.VERSION_1_7) > 0;
       boolean findbugs2orLess = v.getMajor() < 3;
       if (java8orMore && findbugs2orLess) {
-          throw new FindBugsVersionTooLowException("The version of FindBugs (" + v + ") inferred from FindBugs classpath is too low to work with currently used Java version (" + javaVersion + ")."
-                  + " Please use higher version of FindBugs. Inspected FindBugs classpath: " + fileNamesOnClasspath);
+          throw new FindBugsVersionTooLowException("The version of SpotBugs (" + v + ") inferred from SpotBugs classpath is too low to work with currently used Java version (" + javaVersion + ")."
+                  + " Please use higher version of SpotBugs. Inspected SpotBugs classpath: " + fileNamesOnClasspath);
       }
   }
 
@@ -35,9 +35,9 @@ public class FindBugsClasspathValidator {
 
   private VersionNumber getFindbugsVersion(Iterable<String> classpath) {
       for (String f: classpath) {
-          Matcher m = Pattern.compile("(spotbugs|findbugs)-(\\d+.*)\\.jar").matcher(f);
+          Matcher m = Pattern.compile("spotbugs-(\\d+.*)\\.jar").matcher(f);
           if (m.matches()) {
-              return VersionNumber.parse(m.group(2));
+              return VersionNumber.parse(m.group(1));
           }
       }
       throw new GradleException("Unable to infer the version of FindBugs from currently specified FindBugs classpath: " + classpath);

--- a/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
+++ b/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
@@ -35,7 +35,7 @@ public class FindBugsClasspathValidator {
 
   private VersionNumber getFindbugsVersion(Iterable<String> classpath) {
       for (String f: classpath) {
-          Matcher m = Pattern.compile("spotbugs-(\\d+.*)\\.jar").matcher(f);
+          Matcher m = Pattern.compile("spotbugs-(\\d+\\.\\d+\\.\\d+).*\\.jar").matcher(f);
           if (m.matches()) {
               return VersionNumber.parse(m.group(1));
           }

--- a/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
+++ b/gradlePlugin/src/main/java/com/github/spotbugs/internal/spotbugs/FindBugsClasspathValidator.java
@@ -35,9 +35,9 @@ public class FindBugsClasspathValidator {
 
   private VersionNumber getFindbugsVersion(Iterable<String> classpath) {
       for (String f: classpath) {
-          Matcher m = Pattern.compile("findbugs-(\\d+.*)\\.jar").matcher(f);
+          Matcher m = Pattern.compile("(spotbugs|findbugs)-(\\d+.*)\\.jar").matcher(f);
           if (m.matches()) {
-              return VersionNumber.parse(m.group(1));
+              return VersionNumber.parse(m.group(2));
           }
       }
       throw new GradleException("Unable to infer the version of FindBugs from currently specified FindBugs classpath: " + classpath);

